instrumentation/opentelemetry-instrumentation-mysqlclient/tests/test_mysqlclient_integration.py

```diff
@@ -0,0 +1,116 @@
+        with self.disable_logging():
+            MySQLClientInstrumentor().uninstrument()
+
+    @mock.patch("MySQLdb.connect")
+    # pylint: disable=unused-argument
+    def test_instrumentor(self, mock_connect):
+        MySQLClientInstrumentor().instrument()
+
+        cnx = MySQLdb.connect(database="test")
+        cursor = cnx.cursor()
+        query = "SELECT * FROM test"
+        cursor.execute(query)
+
+        spans_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans_list), 1)
+        span = spans_list[0]
+
+        # Check version and name in span's instrumentation info
+        self.assertEqualSpanInstrumentationInfo(
+            span, opentelemetry.instrumentation.mysqlclient
+        )
+
+        # check that no spans are generated after uninstrument
+        MySQLClientInstrumentor().uninstrument()
+
+        cnx = MySQLdb.connect(database="test")
+        cursor = cnx.cursor()
+        query = "SELECT * FROM test"
+        cursor.execute(query)
+
+        spans_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans_list), 1)
+
+    @mock.patch("MySQLdb.connect")
+    # pylint: disable=unused-argument
+    def test_custom_tracer_provider(self, mock_connect):
+        resource = resources.Resource.create({})
+        result = self.create_tracer_provider(resource=resource)
+        tracer_provider, exporter = result
+
+        MySQLClientInstrumentor().instrument(tracer_provider=tracer_provider)
+
+        cnx = MySQLdb.connect(database="test")
+        cursor = cnx.cursor()
+        query = "SELECT * FROM test"
+        cursor.execute(query)
+
+        spans_list = exporter.get_finished_spans()
+        self.assertEqual(len(spans_list), 1)
+        span = spans_list[0]
+
+        self.assertIs(span.resource, resource)
+
+    @mock.patch("MySQLdb.connect")
+    # pylint: disable=unused-argument
+    def test_instrument_connection(self, mock_connect):
+        cnx = MySQLdb.connect(database="test")
+        query = "SELECT * FROM test"
+        cursor = cnx.cursor()
+        cursor.execute(query)
+
+        spans_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans_list), 0)
+
+        cnx = MySQLClientInstrumentor().instrument_connection(cnx)
+        cursor = cnx.cursor()
+        cursor.execute(query)
+
+        spans_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans_list), 1)
+
+    @mock.patch("MySQLdb.connect")
+    # pylint: disable=unused-argument
+    def test_uninstrument_connection(self, mock_connect):
+        MySQLClientInstrumentor().instrument()
+        cnx = MySQLdb.connect(database="test")
+        query = "SELECT * FROM test"
+        cursor = cnx.cursor()
+        cursor.execute(query)
+
+        spans_list = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans_list), 1)
+
+        cnx = MySQLClientInstrumentor().uninstrument_connection(cnx)
```

Comment:
It will be better to clear the spans before the uninstrument, like this:
https://github.com/open-telemetry/opentelemetry-python-contrib/blob/46e4b1da44c534fed8e1002899e9e41e6d668018/instrumentation/opentelemetry-instrumentation-falcon/tests/test_falcon.py#L214

Reply:
Good point, done 👍